core/tracker/tracker.go

```diff
@@ -94,11 +106,14 @@ func (t *Tracker) Run(ctx context.Context) error {
 			t.events[e.duty] = append(t.events[e.duty], e)
 		case duty := <-t.deadliner.C():
 			failed, failedComponent, failedMsg := analyseDutyFailed(duty, t.events[duty])
-
 			t.failedDutyReporter(duty, failed, failedComponent.String(), failedMsg)
 
-			// TODO(dhruv): Case of cluster participation
-			// analyseParticipation(duty, t.events[duty])
+			participatedShares, err := analyseParticipation(t.events[duty])
+			if err != nil {
+				log.Error(ctx, "Invalid participated shares", err)
```

Comment:
btw, you need to add tracker log topic to ctx

Reply:
done